packages/guides/src/Compiler/Passes/GlobalMenuPass.php

```diff
@@ -130,17 +133,31 @@ private function addSubEntries(
             return;
         }
 
-        foreach ($documentEntry->getChildren() as $subDocumentEntryNode) {
-            $subMenuEntry = new InternalMenuEntryNode(
-                $subDocumentEntryNode->getFile(),
-                $subDocumentEntryNode->getTitle(),
-                [],
-                false,
-                $currentLevel,
-                '',
-            );
-            $sectionMenuEntry->addMenuEntry($subMenuEntry);
-            $this->addSubEntries($compilerContext, $subMenuEntry, $subDocumentEntryNode, $currentLevel + 1, $maxDepth);
+        if (!$entryNode instanceof DocumentEntryNode) {
+            return;
+        }
+
+        foreach ($entryNode->getChildren() as $subEntryNode) {
+            if ($subEntryNode instanceof DocumentEntryNode) {
+                $subMenuEntry = new InternalMenuEntryNode(
+                    $subEntryNode->getFile(),
+                    $subEntryNode->getTitle(),
+                    [],
+                    false,
+                    $currentLevel,
+                    '',
+                );
+                $sectionMenuEntry->addMenuEntry($subMenuEntry);
+                $this->addSubEntries($compilerContext, $subMenuEntry, $subEntryNode, $currentLevel + 1, $maxDepth);
+            } elseif ($subEntryNode instanceof ExternalEntryNode) {
```

Comment:
Please separate these blocks as they are not connected, maybe we should extract both strategies to a different method in this class to make it easier to read?

```suggestion
                $this->addSubEntries($compilerContext, $subMenuEntry, $subEntryNode, $currentLevel + 1, $maxDepth);
                continue;
            } 
            
            if ($subEntryNode instanceof ExternalEntryNode) {
```

Reply:
What exactly do you mean with "separate the blocks"?

Reply:
the `if` and `elseif` are called code blocks, basically everything within `{}` is a code block. A refactor strategy is to always extract blocks, some people state that you should only have 1 level of blocks in a method. I do not want to go this far. but as this is like a strategy it would make sense to make it easy to see what is handled. 

Extracting the strategies to different methods helps to read this method, as it would simply read like

```
if (a) -> doSomething
if (b) -> doSomethingElse
```